src/blockdata/script.rs

```diff
@@ -227,18 +243,66 @@ pub fn read_scriptbool(v: &[u8]) -> bool {
 }
 
 /// Read a script-encoded unsigned integer
+///
+/// ## Errors
+///
+/// This function returns an error in these cases:
+///
+/// * `data` is shorter than `size` - `EarlyEndOfScript`
+/// * `size` is greater than `u16::max_value / 8` (8191) - `NumericOverflow`
```

Comment:
nit: an arrow `->` instead of a minus `-` would be more clear to me

Reply:
Good point, I will use `=>` which resembles `match` and thus feels more natural.